admin/public/js/components/charts.jsx

```diff
@@ -0,0 +1,102 @@
+define(['d3', 'react'], function(d3, React) {
+
+  return React.createClass({
+    displayName: 'MovingLineChart',
+
+    getInitialState: function() {
+
+      var that = this;
+      this.x = d3.scale.linear().domain(this.props.xdomain).range([0, this.props.width]);
+      this.y = d3.scale.linear().domain(this.props.ydomain).range([this.props.height, 0]);
+      this.line = d3.svg.line()
+        .interpolate(this.props.interpolation)
+        .x(function(d, i) { return that.x(i); })
+        .y(function(d, i) { return that.y(d); });
```

Comment:
You may consider moving the creation of data-oriented properties on `this.` (`this.x` and `this.y`) into `componentWillMount`, and the creation of DOM nodes into either `componentWillMount` (runs before first render) or `componentDidMount` (runs after first render). `getInitialState` should generally be a pure function that only returns the initial data for `this.state`.